tests/interpreter_tests.rs

```diff
@@ -0,0 +1,402 @@
+                .iter()
+                .map(|(_, bitwidth)| bitwidth)
+                .max()
+                .unwrap()
+                .to_string()
+                .as_str(),
+        )
+        .replace(
+            "{display_outputs}",
+            outputs
+                .iter()
+                .map(|(name, _)| {
+                    format!(
+                        "$display(\"output: %d\\n\", simulate_with_verilator_test_module.{});",
+                        name
+                    )
+                })
+                .collect::<Vec<String>>()
+                .join("\n")
+                .as_str(),
+        );
+
+    let executable_name = "executable";
+    let verilator_output_dir = test_output_dir.join("obj_dir");
+    let executable_path = verilator_output_dir.join(executable_name);
+
+    let default_extra_args = format!(
+        "-I{}",
+        include_dirs
+            .iter()
+            .map(|path| path.to_str().unwrap())
+            .collect::<Vec<&str>>()
+            .join(" -I")
+    );
+
+    let makefile_prog = std::fs::read_to_string(makefile_template_path)
+        .unwrap()
+        .replace("{testbench_file_path}", testbench_path.to_str().unwrap())
+        .replace(
+            "{verilator_output_dir}",
+            verilator_output_dir.to_str().unwrap(),
+        )
+        .replace("{simulation_executable_name}", executable_name)
+        .replace("{extra_verilator_args}", &default_extra_args);
+
+    std::fs::write(&testbench_path, &testbench_prog).unwrap();
+    std::fs::write(&makefile_path, &makefile_prog).unwrap();
+
+    // TODO(@ninehusky): We can get rid of the necessity for a Makefile after this PR is merged
+    // into Verilator: https://github.com/verilator/verilator/pull/5031
+    let verilator_compile_output = std::process::Command::new("make")
+        .arg("--environment-overrides")
+        .arg("--always-make")
+        .arg("-f")
+        .arg(makefile_path)
+        .output()
+        .unwrap();
+
+    if !verilator_compile_output.status.success() {
+        panic!(
+            "Verilator failed, stderr: {:?}",
+            String::from_utf8(verilator_compile_output.stderr)
+        );
+    }
+
+    let mut sim_proc = std::process::Command::new(executable_path)
+        .stdin(std::process::Stdio::piped())
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let sim_proc_stdin = sim_proc.stdin.as_mut().unwrap();
+
+    let num_inputs = inputs.len();
+
+    // TODO(@ninehusky): this is going to assume we only want to interpret on the first output included in `outputs`.
+    let (serialized, root_node) = prep_interpreter(
+        verilog_module_path.clone(),
+        test_output_dir.clone(),
+        top_module_name,
+        outputs[0].0,
+    );
+
+    sim_proc_stdin
+        .write_all(format!("{} {} {}\n", num_inputs, num_test_cases, num_clock_cycles).as_bytes())
+        .unwrap();
+
+    let mut rng = rand::thread_rng();
```

Comment:
Let's make sure to seed the RNG with a constant for now, for reproducibility.

Reply:
Done!